src/main/java/net/snowflake/client/jdbc/ArrowResultChunk.java

```diff
@@ -589,8 +589,8 @@ private void mergeBatchesIntoOne() throws SnowflakeSQLException
     }
     catch (SFException ex)
     {
-      throw new SnowflakeSQLException(ex, SqlState.INTERNAL_ERROR,
-                                      ErrorCode.INTERNAL_ERROR.getMessageCode(),
+      throw new SnowflakeSQLLoggedException(ex, SqlState.INTERNAL_ERROR,
+                                      ErrorCode.INTERNAL_ERROR.getMessageCode(), null,
```

Comment:
A session must be available when an arrow result chunk is created. You probably need to change the constructor to have a session object and pass here.